erts/preloaded/src/socket.erl

```diff
@@ -2787,22 +2787,20 @@ enc_type(_, raw)       -> ?SOCKET_TYPE_RAW;
 enc_type(_, seqpacket) -> ?SOCKET_TYPE_SEQPACKET;
 enc_type(_, Type)      -> invalid_type(Type).
 
--spec enc_protocol(Type, Protocol) -> non_neg_integer() | 
-                                      {raw, non_neg_integer()} when
-      Type     :: type(),
+-spec enc_protocol(Protocol) -> non_neg_integer() |
+                                {raw, non_neg_integer()} when
       Protocol :: protocol().
 
-enc_protocol(_,         default) -> ?SOCKET_PROTOCOL_DEFAULT;
-enc_protocol(dgram,     ip)      -> ?SOCKET_PROTOCOL_IP;
-enc_protocol(stream,    tcp)     -> ?SOCKET_PROTOCOL_TCP;
-enc_protocol(dgram,     udp)     -> ?SOCKET_PROTOCOL_UDP;
-enc_protocol(seqpacket, sctp)    -> ?SOCKET_PROTOCOL_SCTP;
-enc_protocol(dgram,     icmp)    -> ?SOCKET_PROTOCOL_ICMP;
-enc_protocol(raw,       icmp)    -> ?SOCKET_PROTOCOL_ICMP;
-enc_protocol(raw,       igmp)    -> ?SOCKET_PROTOCOL_IGMP;
-enc_protocol(raw,       {raw, P} = RAW) when is_integer(P) -> RAW;
-enc_protocol(Type, Proto) -> 
-    invalid_protocol(Type, Proto).
+enc_protocol(default) -> ?SOCKET_PROTOCOL_DEFAULT;
+enc_protocol(ip)      -> ?SOCKET_PROTOCOL_IP;
+enc_protocol(tcp)     -> ?SOCKET_PROTOCOL_TCP;
+enc_protocol(udp)     -> ?SOCKET_PROTOCOL_UDP;
+enc_protocol(sctp)    -> ?SOCKET_PROTOCOL_SCTP;
+enc_protocol(icmp)    -> ?SOCKET_PROTOCOL_ICMP;
+enc_protocol(igmp)    -> ?SOCKET_PROTOCOL_IGMP;
+enc_protocol({raw, P} = RAW) when is_integer(P) -> RAW;
+enc_protocol(Proto) ->
+    invalid_protocol(Proto).
```

Comment:
There is a call to `enc_protocol(Type, Protocol)` in `open/4` that needs to be changed too.  This does not compile.